apps/src/blockly/addons/functionEditor.ts

```diff
@@ -355,61 +379,68 @@ export default class FunctionEditor {
       hiddenProcedure
     );
 
-    this.editorWorkspace.getProcedureMap().add(editorProcedureModel);
+    this.editorWorkspace?.getProcedureMap().add(editorProcedureModel);
     Blockly.Events.enable();
 
     this.showForFunction(hiddenProcedure, procedureType);
   };
 
   onDeletePressed() {
+    if (!this.block) {
+      return;
+    }
     /// We use "delete" for cancel and "keep" for confirm so that the
     // delete button is red and the keep button is purple.
     Blockly.customSimpleDialog({
-      bodyText: msg.confirmDeleteFunctionWarning({
+      bodyText: commonI18n.confirmDeleteFunctionWarning({
         functionName: this.block.getProcedureModel().getName(),
       }),
-      cancelText: msg.delete(),
+      cancelText: commonI18n.delete(),
       isDangerCancel: true, // gives us a red button
-      confirmText: msg.keep(),
+      confirmText: commonI18n.keep(),
       onConfirm: null, // No-op
       onCancel: this.onDeleteConfirmed.bind(this),
     });
   }
 
   onDeleteConfirmed() {
+    if (!this.block) {
+      return;
+    }
     // delete all caller blocks from the procedure workspace
     Blockly.Procedures.getCallers(
       this.block.getProcedureModel().getName(),
       Blockly.getHiddenDefinitionWorkspace()
     ).forEach(block => {
-      block.dispose();
+      block.dispose(true /* healStack */);
     });
 
     // delete all caller blocks from the main workspace
     Blockly.Procedures.getCallers(
       this.block.getProcedureModel().getName(),
       Blockly.mainBlockSpace
     ).forEach(block => {
-      block.dispose();
+      block.dispose(true /* healStack */);
     });
 
     // delete the block from the editor workspace's procedure map
     // this will also cause it to be deleted from the main and procedure
     // workspaces' map
     this.editorWorkspace
-      .getProcedureMap()
+      ?.getProcedureMap()
       .delete(this.block.getProcedureModel().getId());
 
-    // delete the block from the editor workspace and hide the modal
-    this.block.dispose();
+    // Delete the block from the editor workspace and hide the modal.
+    // We don't need to heal the stack here because we always clear the editor workspace.
+    this.block.dispose(false /* healStack */);
```

Comment:
If we don't need to heal the stack, do we need to change this at all?

Reply:
Yeah, technically the parameter is required.

Reply:
Oh, that makes sense!